lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -1393,6 +1420,16 @@ class SemanticsObject {
     }
   }
 
+  /// Recursively visits the tree rooted at `this` node in depth-first fashion.
+  ///
+  /// Calls the [callback] for `this` node, then for all of its descendants.
+  void visitDepthFirst(void Function(SemanticsObject) callback) {
+    callback(this);
```

Comment:
Should `callback` return a `bool` so you abort the the drilling into children if `callback(this)` returns `false`?

Reply:
See https://github.com/flutter/engine/pull/41681#discussion_r1189263649